packages/server/src/plugin/cacheControl/index.ts

```diff
@@ -48,6 +48,14 @@ export interface ApolloServerPluginCacheControlOptions {
    * delivery response, and the body contains no errors.
    */
   calculateHttpHeaders?: boolean | 'if-cacheable';
+
+  /**
+   * Disables instrumenting all fields for dynamic cache control hints. This is
+   * a pretty big significant performant boost, especially the larger a response
+   * is.
+   */
+  staticOnly?: boolean;
```

Comment:
@andrewmcgivery What are your thoughts on also supporting a conditional function for the static only option, which provides the option to disable dynamic cache control hints based on the operation?

So, for example:

```
staticOnly?: boolean | ((requestContext: GraphQLRequestContextExecutionDidStart<unknown>) => boolean);
```

Reply:
🤔 it sounds technically feasible... I'll defer to @glasser to get his thoughts!

Reply:
One relevant thing I'm not clear on — as you noted, there's a comment in requestPipeline.ts (which you could update?) noting that if you completely avoid instrumenting the schema entirely by disabling field-level instrumentation in the usage reporting plugin and disabling cache control, then you get a performance boost.  Is that the only performance boost you've been able to measure here: the performance boost of *entirely* avoiding schema instrumentation? ie, if you're testing this in a subgraph, have you either disabled the inline trace plugin or are not running behind a router that ever requests ftv1 inline traces?

ie, is the performance boost from "no plugin *ever* returns a `willResolveField` handler so the schema never gets instrumented", or is there still a performance boost from "the schema is instrumented but the cache control plugin never puts a field handler into it"?

If the former, we probably would need to advise that this optimization really only works in conjunction with avoiding field-level instrumentation... and there'd be no point to making this be dynamic per request.  If the latter, then that's simpler.

Reply:
What I found was that just the presence of a `willResolveField` hook in a plugin, even if it a completely empty function, causes a performance hit.

Reply:
I guess my question is (and maybe you can test this) — let's say the plugin is written as "on the very first time the plugin is called on program startup, return a `willResolveField` hook, but then never again return that hook". Do you still see the performance hit?  (ie, is the performance hit caused by the fact that it triggered `enablePluginsForSchemaResolvers` to be called at all, whether or not the hook exists on the particular request being processed?)

Reply:
Actually lemme take a step back. I'm confused about if this PR actually works at all.  The idea behind `staticOnly: true` is that you don't care about `info.cacheControl` stuff but you do want `@cacheControl` on fields and types in the schema to take effect, right?  But the way to that the plugin knows which fields and types are relevant to the operation is based on which fields and types are executed — it all happens inside `willResolveField`.  As far as I understand, with this implementation, `staticOnly: true` also completely disables `@cacheControl` directive processing.  Is that your intention? Or am I confused and it still works somehow?